spec/unit/full_transmission_spec.rb

```diff
@@ -48,9 +49,14 @@
           loop do
             batch = transmit_queue.pop(true) rescue break
             begin
-              Timeout.timeout(5) do
```

Comment:
review note: `Timeout.timeout` doesn't raise `Execution expired` exception with LS 7.x (I honestly couldn't figure out why). This change simulates _exact_ `timeout` behavior and proves that internal logics are fine.